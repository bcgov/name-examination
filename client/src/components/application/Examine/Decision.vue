/* eslint-disable */
<template>
  <span>
    <div class="row">
      <div class="col-6 add-top-padding">
        <h3>Conditions</h3>
        <div>
        <multiselect
          v-model="conditions_selected"
          :options="conditions_onlycustomerfacing"
          :multiple="true"
          :custom-label="conditionsLabel"
          name="phrase"
          track-by="phrase"
          :close-on-select="true"
          deselectLabel=""
          selectLabel=""
          selectedLabel=""
          placeholder=""
          :disabled="customer_message_override !== null"
        />
        </div>
      </div>
      <div class="col-6 add-top-padding">
        <h3>Conflicts</h3>
        <multiselect
          v-model="conflicts_selected"
          :options="conflictList"
          :multiple="true"
          :custom-label="conflictsLabel"
          track-by="nrNumber"
          :max="3"
          :close-on-select="true"
          deselectLabel=""
          selectLabel=""
          selectedLabel=""
          placeholder=""
          :disabled="customer_message_override !== null"
        />
      </div>
    </div>

    <div class="row">
      <div class="col-6 add-top-padding">
        <h3>Generic Rejection Reason (label TBD)</h3>
        <multiselect
          v-model="decision_reasons_selected"
          :options="listDecisionReasons"
          :multiple="true"
          label="name"
          track-by="name"
          :close-on-select="true"
          deselectLabel=""
          selectLabel=""
          selectedLabel=""
          placeholder=""
          :disabled="customer_message_override !== null"
        />
      </div>
      <div class="col-6 add-top-padding">
        <h3>Trademarks</h3>
        <div>
        <multiselect
          v-model="trademarks_selected"
          :options="trademarks"
          :multiple="true"
          :custom-label="trademarksLabel"
          name="name"
          track-by="name"
          :close-on-select="true"
          deselectLabel=""
          selectLabel=""
          selectedLabel=""
          placeholder=""
          :disabled="customer_message_override !== null"
        />
        </div>
      </div>
    </div>

    <!-- internal alerts -->
    <div class="row">
      <div class="col add-top-padding">
        <div class="alert alert-warning"
             v-for="record in conditions_onlyinternalinstructions" v-bind:key="record.id">
            {{ record.text }}
        </div>
      </div>
    </div>

    <div class="row">
      <div class="col add-top-padding">
        <h3>Message to Requestor</h3>
        <div class="customer-msg-box">
          <pre>{{ customer_message_display }}</pre>
        </div>

        <!-- character count warning -->
        <div v-bind:class="{ 'warning': customer_message_display.length > 955,
                              'character_count': true }">
          Character count: {{ customer_message_display.length }}
          <span v-if="customer_message_display.length > 955" style="color: red;">
            <i class="fa fa-warning"></i>
            The message will be cut off at 955 characters.
          </span>
        </div>


        <a href="#" id="edit-customer-message-link" data-toggle="modal"
           data-target="#edit-customer-message-modal">Edit...</a>
        <a href="#" id="clear-customer-message-link" v-if="customer_message_override"
           @click="clearCustomerMessagOverride">Clear Customer Message Override</a>
      </div>
    </div>

    <internalcomments ref="decisioncomments" layout="decision"></internalcomments>

    <!-- Edit Customer Comments popup -->
    <div class="modal fade" id="edit-customer-message-modal" tabindex="-1" role="dialog">
      <div class="modal-dialog modal-lg" role="document">
        <div class="modal-content">
          <div class="modal-header">
            <h5 class="modal-title" id="exampleModalLabel">Edit Customer Comments</h5>
            <button type="button" class="close" data-dismiss="modal" aria-label="Close">
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div class="modal-body">
            <textarea id="edit-customer-message-textarea" class="form-control" rows="10"
                      v-model="customer_message_display"></textarea>
          </div>
          <div class="modal-footer">
            <button type="button" class="btn btn-sm btn-secondary"
                    data-dismiss="modal">Cancel</button>
            <button type="button" class="btn btn-sm btn-primary"
                    @click="saveCustomerMessageOverride">Save</button>
          </div>
        </div>
      </div>
    </div>
  </span>
</template>

<script>
/* eslint-disable */

  import internalcomments from "@/components/application/Examine/InternalComments.vue";
  import Multiselect from 'vue-multiselect';

  const myMixin = {
    created(){
      console.log("It works!")
    }
  }

  export default {
    name: "Decision",
    mixins: [myMixin],
    data: function ()
    {
      return {
        conditions_selected: [],
        conflicts_selected: [],
        decision_reasons_selected: [],
        trademarks_selected: [],
        customer_message_override: null,
      }
    },
    computed: {
      acceptance_will_be_conditional() {
        return this.$store.getters.acceptance_will_be_conditional;
      },
      is_making_decision: {
        get: function() {
          return this.$store.getters.is_making_decision;
        },
        set: function(value) {
          this.$store.commit('is_making_decision', value);
        }
      },
      decision_made: {
        get: function () {
          return this.$store.getters.decision_made;
        },
        set: function (value) {
          this.$store.commit('decision_made', value);
        }
      },
      currentNameObj: {
        get: function () {
          return this.$store.getters.currentNameObj;
        },
        set: function (value) {
          this.$store.dispatch('currentNameObj', value);
        }
      },
      conflictList() {
        return this.$store.getters.conflictList;
      },
      currentConflict() {
        return this.$store.getters.currentConflict;
      },
      listDecisionReasons() {
        return this.$store.getters.listDecisionReasons;
      },
      conditions() {
        /* Re-arrange the fields in object to be easier to use in dropdowns.

        In the case of multiple conditions per word, break each condition into its own row for the
        dropdown (the nested for loops below). At time of development there is no data that actually
        has multiple conditions per word.

        Note: in the case where a word does not have any conditions (ie: nothing in the conditions
        xref table) we never get to the point where we create the revised_match_object, and never
        push it to the array. It is as if they were not returned.
        */

        try {
          if (this.$store.getters.conditionsJSON.restricted_words_conditions !== undefined) {
            var condition_matches = this.$store.getters.conditionsJSON.restricted_words_conditions;
            var retval = []

            for (let wordmatch of condition_matches) {
             for (let condition of wordmatch.cnd_info) {
               var revised_match_object = {};
               revised_match_object.id = wordmatch.word_info.id;
               revised_match_object.phrase = wordmatch.word_info.phrase;
               revised_match_object.allow_use = (condition.allow_use == 'Y')?true:false;
               revised_match_object.consent_required = (condition.consent_required == 'Y')?true:false;
               revised_match_object.instructions = condition.instructions;
               revised_match_object.text = condition.text;

               retval.push(revised_match_object);
             }
            }

            return retval;

          }
          else return []
        } catch (err) {
          return [];
        }
      },
      conditions_onlycustomerfacing() {
        /* subset of conditions, only those that have customer instructions
        */

        var arr_conditions = this.conditions.filter(function (el) {
          return el.instructions !== '' && el.instructions !== null
        });

        // manually add "CONSENT REQUIRED" condition
        arr_conditions.push({
          instructions: "Consent Required.",
          consent_required: true,
          display_string: "Consent Required",
        });

        return arr_conditions;

      },
      conditions_onlyinternalinstructions() {
        return this.conditions.filter(function (el) {return el.text !== '' && el.text !== null});

      },
      currentCondition() {
        var currentCondition = this.$store.getters.currentCondition;
        if (currentCondition == null) return null;

        // only use the currentCondition if it requires consent
        if (currentCondition.consent_required != 'Y') return null;

        // Re-arrange the fields in object to be easier to use in dropdowns - match main list
        var revised_match_object = {};
        revised_match_object.phrase = currentCondition.word;
        revised_match_object.allow_use = (currentCondition.allow_use == 'Y')?true:false;
        revised_match_object.consent_required = (currentCondition.consent_required == 'Y')?true:false;
        revised_match_object.instructions = currentCondition.instructions;
        revised_match_object.text = currentCondition.text;

        return revised_match_object;

      },
      trademarks() {
        try {
          if (this.$store.getters.trademarksJSON !== null) {
            return this.$store.getters.trademarksJSON.names;
          }
          else return []
        } catch (err) {
          return [];
        }
      },
      currentTrademark() {
        return this.$store.getters.currentTrademark;
      },
      customer_message() {
        /*
        Build customer message based on selected conditions, conflicts, trademarks, and format.
         */

        var retval = [];

        // CONFLICTS
        for (var i = 0; i < this.conflicts_selected.length; i++) {
          retval.push('Rejected due to conflict with ' + this.conflicts_selected[i].text);
        }

        // CONDITIONS
        for (var i = 0; i < this.conditions_selected.length; i++) {
          if (this.conditions_selected[i].phrase !== undefined && this.conditions_selected[i].phrase !== '') {
            retval.push(this.conditions_selected[i].phrase + ' - ' + this.conditions_selected[i].instructions);
          }
          else {
            retval.push(this.conditions_selected[i].instructions);
          }
        }

        // TRADEMARKS
        for (var i = 0; i < this.trademarks_selected.length; i++) {
          retval.push('Registered Trademark: ' + this.trademarks_selected[i].name + ' - Application #' + this.trademarks_selected[i].application_number);
        }

        // GENERIC DECISION REASONS
        for (var i = 0; i < this.decision_reasons_selected.length; i++) {
          retval.push(this.decision_reasons_selected[i].reason);
        }

        return retval;

      },
      customer_message_display: {
        get: function () {
          /*
          The text version of customer_message that is sent to API to be stored in DB.
           */

          // if there's a customer message override (ie: the examiner has edited the text) then use
          // that...
          if (this.customer_message_override) {
            return this.customer_message_override;

          } else {
            // otherwise build out formatted text with line breaks...
            var retval = '';

            for (var i = 0; i < this.customer_message.length; i++) {
              retval += this.customer_message[i] + '\n\n';
            }

            return retval;

          }
        },
        set: function (value) {
          /*
          This sets the message override from editing in the popup, NOT customer_message_display
          calculated field.
           */
          //this.customer_message_override = value;

        }
      }
    },
    components: {
      internalcomments,
      Multiselect,
    },
    mounted: function () {
      // pre-select the conflict from the display screen
      if (this.currentConflict !== null && this.currentConflict !== undefined) {
        this.conflicts_selected.push(this.currentConflict);
      }

      // pre-select the condition from the display screen
      if (this.currentCondition !== null && this.currentCondition !== undefined) {
        this.conditions_selected.push(this.currentCondition);
      }


      // pre-select the trademark from the display screen
      if (this.currentTrademark !== null && this.currentTrademark !== undefined) {
        this.trademarks_selected.push(this.currentTrademark);
      }

      // reset the conditional acceptance flag
      this.$store.commit('acceptance_will_be_conditional', false);


    },
    watch: {
      decision_made: function() {
        this.nameAcceptReject();
      },
      conditions_selected: function () {
        // set state variable indicating whether acceptance will be conditional or not

        var retval = false;
        for (var i = 0; i < this.conditions_selected.length; i++) {
          if (this.conditions_selected[i].consent_required) {
            retval = true;
            break;
          }
        }

        this.$store.commit('acceptance_will_be_conditional', retval);
      },
    },
    methods: {
      clearCustomerMessagOverride() {
        this.customer_message_override = null;
      },
      saveCustomerMessageOverride(obj) {
        console.log('save customer message override...');
        console.log(obj);
        this.customer_message_override = $('#edit-customer-message-textarea').val();

        $('#edit-customer-message-modal').modal('hide');
      },
      conflictsLabel(obj) {
        return obj.text + ' - ' + obj.nrNumber;
      },
      conditionsLabel(obj) {
        if (obj.display_string !== undefined) return obj.display_string;
        else return obj.phrase + ' - ' + obj.instructions;
      },
      trademarksLabel(obj) {
        return obj.name + ' - ' + obj.application_number;
      },
      nameAcceptReject() {
        // save decision text, state, decision comment, and up to three conflicts


        if (this.decision_made == 'APPROVED') {
          this.currentNameObj.state = 'APPROVED'; // accepted

          // conditionally accepted if any conditions selected with condition_required flag TRUE
          for (var i = 0; i < this.conditions_selected.length; i++) {
            var record = this.conditions_selected[i];
            if (record.consent_required) {
              this.currentNameObj.state = 'CONDITION';
              break;
            }
          }

          // if there were conflicts selected but this is an approval, this will result in
          // accidental "rejected due to conflict" messaging. Remove it by clearing the selected
          // conflicts (Issue #767).
          this.conflicts_selected = [];
        }
        else {
          this.currentNameObj.state = 'REJECTED';
        }
        for (var i = 0; i < this.conflicts_selected.length; i++) {
          if (i == 0) {
            this.currentNameObj.conflict1 = this.conflicts_selected[i].text;
            this.currentNameObj.conflict1_num = this.conflicts_selected[i].nrNumber;
          }
          if (i == 1) {
            this.currentNameObj.conflict2 = this.conflicts_selected[i].text;
            this.currentNameObj.conflict2_num = this.conflicts_selected[i].nrNumber;

          }
          if (i == 2) {
            this.currentNameObj.conflict3 = this.conflicts_selected[i].text;
            this.currentNameObj.conflict3_num = this.conflicts_selected[i].nrNumber;

          }
        }

        this.currentNameObj.decision_text = this.customer_message_display.substr(0,955);

        // add new comment
        this.$refs.decisioncomments.addNewComment();

        // send decision to API and reset flags
        this.$store.dispatch('nameAcceptReject');
        this.decision_made = null;
        this.is_making_decision = false;
      },
    },
  }



</script>

<style src="vue-multiselect/dist/vue-multiselect.min.css"></style>

<style scoped>

  .customer-msg-box {
    padding: 10px;
    border: 1px solid #bbb;
  }

  .customer-msg-box > p {
    margin-bottom: 10px;
  }

  #edit-customer-message-link,
  #clear-customer-message-link {
    font-size: 12px;
    float: right;
  }

  #clear-customer-message-link {
    margin-right: 20px;
  }

  .customer-msg-box pre {
    font-family: 'Avenir', Helvetica, Arial, sans-serif;
    white-space: pre-line;
  }

</style>

<!-- unscoped styles -->
<style>

  .multiselect, .multiselect__input, .multiselect__single {
    font-size: 12px;
  }

  .multiselect__option {
    min-height: inherit;
    padding: 5px;
  }

  .character_count {
    float: left;
    padding: 5px;
  }
  .character_count.warning {
    background-color: #ffa;
  }


</style>
