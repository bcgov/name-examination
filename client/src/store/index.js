/* eslint-disable */
import Vue from 'vue'
import Vuex from 'vuex'
import axios from '../axios-auth'
import globalAxios from 'axios'
import router from '@/router'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    nrNum: null,
    authorized: null,
    userId: null,
    email: null,
    kctoken: null,
    nr_conflict: null,
    user_role: null,
    currentChoice: null,
    currentName: null,
    currentState: null,
    is_editing: false,
    nrData: null,
    details: null,
   // formData: {
      compInfo: {
        nrNumber: null,
        compNames: {
          compName1: null,
          compName2: null,
          compName3: null
        },
        requestType: null,
      },
      applicantInfo: {
        applicantname: {
          firstName: null,
          lastname: null
        },
        contactInfo: {
          address: null,
          contactName: null,
          phone: null,
          email: null,
          fax: null
        },
      },
      additionalCompInfo: {
        jurisdiction: null,
        natureOfBussiness: null,
        nuans: null,
        sk_name: null,
        nr_status: null
      },
      examiner: null,
      priority: null,
      reSubmission: {
        reSubmissionYN: null,
        linkedNR: null
      },
      reservationCount: null,
      expiryDate: null,
  //  },
    issueText: null,
    issue: {
        issue_Match: null,
        issue_Consent: null,
        issue_TradeMark: null,
        issue_History: null,
        issue_Format: null,
        issue_Match_Text: null,
        issue_Consent_Text: null,
        issue_TradeMark_Text: null,
        issue_History_Text: null,
        issue_Format_Text: null
      }
  },

  mutations: {
    authUser (state, userData) {
      state.kctoken = userData
      //state.kctoken = userData.client_session
      //state.userRole = userData.user_role
      //state.idToken = userData.client_session
    },

    clearAuthData (state) {
      state.kcToken = null
      state.userId = null
    },

    setIssueText (state,issue) {
      switch (issue) {
        case "M1":
          state.issueText = state.issue.issue_Match_Text;
          break;
        case "M2":
          state.issueText = state.issue.issue_Consent_Text;
          break;
        case "M3":
          state.issueText = state.issue.issue_TradeMark_Text;
          break;
        case "M4":
          state.issueText = state.issue.issue_History_Text;
          break;
        case "M5":
          state.issueText = state.issue.issue_Format_Text;
          break;
        default:
          state.issueText = state.issue.issue_Match_Text;
      }
    },

    loadpostgresNo(state, postgresData) {
        state.compInfo.nrNumber = postgresData.nameRequest
    },

    loadCompanyInfo(state, dbcompanyInfo) {
      console.log('Loading Company Info into state')

      state.nrData = dbcompanyInfo

      if (dbcompanyInfo.names.length == 0) {
        console.log('Error, No company names found')
        return
      } else {
        for (let record of dbcompanyInfo.names) {
          switch (record.choice) {
            case 1:
              state.compInfo.compNames.compName1 = record.name
              break;
            case 2:
              state.compInfo.compNames.compName2 = record.name
              break;
            case 3:
              state.compInfo.compNames.compName3 = record.name
              break;
            default:
              console.log('Error with company name structure')
              return
          }
        }
      }

      state.compInfo.requestType = dbcompanyInfo.requestTypeCd
      state.applicantInfo.applicantname.firstName = dbcompanyInfo.applicant
      //state.applicantInfo.applicantname.lastname = dbcompanyInfo.lastName
      //state.applicantInfo.contactInfo.address = dbcompanyInfo.address
      state.applicantInfo.contactInfo.contactName = dbcompanyInfo.contact
      state.applicantInfo.contactInfo.phone = dbcompanyInfo.phoneNumber
      //state.applicantInfo.contactInfo.email = dbcompanyInfo.email
      //state.applicantInfo.contactInfo.fax = dbcompanyInfo.fax
      state.additionalCompInfo.jurisdiction = dbcompanyInfo.xproJurisdiction
      state.additionalCompInfo.natureOfBussiness = dbcompanyInfo.natureBusinessInfo
      state.additionalCompInfo.nuans = dbcompanyInfo.nuansNum
      state.additionalCompInfo.sk_name = dbcompanyInfo.skPartner
      state.additionalCompInfo.nr_status = dbcompanyInfo.state
      state.examiner = dbcompanyInfo.userId
      state.priority = dbcompanyInfo.priorityCD
      //state.reSubmission.reSubmissionYN = dbcompanyInfo.resubmissionYN
      //state.reSubmission.linkedNR = dbcompanyInfo.linkedNR
      //state.reservationCount = dbcompanyInfo.reservationCount
      state.expiryDate = dbcompanyInfo.expiryDate
    },

    loadCompanyIssues(state, dbcompanyIssues) {
      //state.issue.issue_Match = dbcompanyIssues.isue_Match
      //state.issue.issue_Match_Text = dbcompanyIssues.issue_Match_Text
      //state.issue.issue_Consent = dbcompanyIssues.issue_Consent
      //state.issue.issue_Consent_Text = dbcompanyIssues.issue_Consent_Text
      //state.issue.issue_TradeMark = dbcompanyIssues.issue_TradeMark
      //state.issue.issue_TradeMark_Text = dbcompanyIssues.issue_TradeMark_Text
      //state.issue.issue_History = dbcompanyIssues.issue_History
      //state.issue.issue_History_Text = dbcompanyIssues.issue_History_Text
      //state.issue.issue_Format = dbcompanyIssues.issue_Format
      //state.issue.issue_Format_Text = dbcompanyIssues.issue_Format_Text
    },

    update_nrData(state){

      if(state.nrData.names.length == 0) {
        console.log('Error, No company names found')
        return
      } else {
        for (let record of state.nrData.names) {
          switch (record.choice) {
            case 1:
              state.nrData.names[0].name = state.compInfo.compNames.compName1
              break;
            case 2:
              state.nrData.names[0].name = state.compInfo.compNames.compName2
              break;
            case 3:
              state.nrData.names[0].name = state.compInfo.compNames.compName3
              break;
          }
        }
      }

      state.nrData.requestTypeCd = state.compInfo.requestType
      state.nrData.applicant = state.applicantInfo.applicantname.firstName
      //state.applicantInfo.applicantname.lastname = dbcompanyInfo.lastName
      //state.applicantInfo.contactInfo.address = dbcompanyInfo.address
      state.nrData.contact = state.applicantInfo.contactInfo.contactName
      state.nrData.phoneNumber =  state.applicantInfo.contactInfo.phone
      //state.applicantInfo.contactInfo.email = dbcompanyInfo.email
      //state.applicantInfo.contactInfo.fax = dbcompanyInfo.fax
      state.nrData.xproJurisdiction = state.additionalCompInfo.jurisdiction
      state.nrData.natureBusinessInfo = state.additionalCompInfo.natureOfBussiness
      state.nrData.nuansNum = state.additionalCompInfo.nuans
      state.nrData.skPartner = state.additionalCompInfo.sk_name
      //state.nrData.state =  state.additionalCompInfo.nr_status
      state.nrData.userId = state.examiner
      state.nrData.priorityCD = state.priority
      //state.reSubmission.reSubmissionYN = dbcompanyInfo.resubmissionYN
      //state.reSubmission.linkedNR = dbcompanyInfo.linkedNR
      //state.reservationCount = dbcompanyInfo.reservationCount
      state.nrData.expiryDate = state.expiryDate
    },

    saveDetail(state,detail){
      state.details = detail
    }
  },

  actions: {
    kcauth ({commit, dispatch, state}) {
      const kc = localStorage.getItem('KEYCLOAK_TOKEN')
      state.kctoken = kc
      //console.log(authData)
      //state.user_role =  authData.realm_access.roles,
      state.user_role = ''
      state.idToken = kc
      localStorage.setItem('kctoken',state.kctoken)
      localStorage.setItem('user_role',state.user_role)

      if (state.kctoken){
        console.log('KC Authorized user roles')
        commit('authUser', {
          //user_role: state.user_role,
          kctoken: state.kctoken
        })
      }
    },

    logout ({commit,state}) {

      commit('clearAuthData')

      localStorage.removeItem('expirationDate')
      localStorage.removeItem('token')
      localStorage.removeItem('user_role')
      localStorage.removeItem('userId')
      localStorage.removeItem('email')
      localStorage.removeItem('COMPINFO')

      localStorage.removeItem('kctoken')
      localStorage.removeItem('KEYCLOAK_TOKEN')
      localStorage.removeItem('AUTHORIZED')
      localStorage.removeItem('USERNAME')

    },

    tryAutoLogin ({commit}) {
    },

    storeUser ({commit, state}, userData) {
      if (!state.idToken) {
        return
      }
      globalAxios.post('/users.json' + '?auth=' + state.idToken, userData)
        .then(res => console.log(res))
        .catch(error => console.log(error))
    },

    fetchUser ({commit, state}) {
      if (!state.idToken) {
        return
      }
      globalAxios.get('/users.json' + '?auth=' + state.idToken)
        .then(res => {
          console.log(res)
          const data = res.data
          const users = []
          for (let key in data) {
            const user = data[key]
            user.id = key
            users.push(user)
          }
          console.log(users)
          commit('storeUser', users[0])
        })
        .catch(error => console.log(error))
    },

    setDetails({commit, state}) {
     var detail = state.details?null:"1";
     commit('saveDetail',detail)
    },

    selectNameIssue ({commit, dispatch, state}, divID) {
      console.log('action: selected issue control:' + divID)
      localStorage.setItem('issueText',divID )
      commit('setIssueText', divID)
    },

    getpostgrescompNo ({commit, dispatch, state}) {
      console.log('action: select next company number from postgres')
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/queues/@me/oldest'
      const vm = this
      return axios.get(url, {headers: {Authorization: `Bearer ${myToken}`}}).then(response => {
        console.log('Comp No Response:' + response)
        commit('loadpostgresNo',response.data)
      })
    },

    getpostgrescompInfo ({commit},nrNumber) {
      console.log('action: getting data for company number: ' + nrNumber + ' from postgres')
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/' + nrNumber
      console.log('URL:' + url)
      const vm = this
      return axios.get(url, {headers: {Authorization: `Bearer ${myToken}`}}).then(response => {
        console.log('Comp Info Response:' + response.data)
        localStorage.setItem('COMPINFO',response.data)
        commit('loadCompanyInfo',response.data)
      })
      .catch(error => console.log('ERROR: ' + error))
    },

    updateNRState ({state},nrState) {
      console.log('Updating Examination state for number: ' + state.compInfo.nrNumber)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber

      axios.patch(url,{headers: {Authorization: `Bearer ${myToken}`}},{"state": nrState} )
           .then(function(response){
                console.log('state updated to ' + nrState + ' for ' + state.compInfo.nrNumber)
            })
            .catch(error => console.log('ERROR: ' + error))
      },

    nameApproved( {commit, state}) {
      console.log('Name Approved for ' + state.compInfo.nrNumber + ", " + state.currentName)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      state.nrData.state = 'APPROVED'
      commit('update_nrData')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber
      axios.put(url, state.nrData, {headers: {Authorization: `Bearer ${myToken}`}})
           .then(function(response){
              console.log('Name approved for ' + state.compInfo.nrNumber)
           })
           .catch(error => console.log('ERROR: ' + error))
    },

    nameRejected({commit, state}) {
      console.log('Name Rejected for NR #' + state.compInfo.nrNumber)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      state.nrData.state = 'REJECTED'
      commit('update_nrData')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber
      axios.put(url,state.nrData, {headers: {Authorization: `Bearer ${myToken}`}})
           .then(function(response){
              console.log('Name rejected for ' +  state.compInfo.nrNumber)
            })
            .catch(error => console.log('ERROR: ' + error))
    }
  },

  getters: {
    is_editing(state) {
      return state.is_editing
    },
    user(state) {
      return state.user
    },
    email(state) {
      return state.email
    },
    issueText(state) {
      return state.issueText
    },
    isAuthenticated(state) {
      return localStorage.getItem("AUTHORIZED")
    },
    nrNumber(state) {
      return state.compInfo.nrNumber
    },
    compName1(state) {
     return state.compInfo.compNames.compName1
    },
    compName2(state) {
      return state.compInfo.compNames.compName2
    },
    compName3(state) {
      return state.compInfo.compNames.compName3
    },
    requestType(state) {
      return state.compInfo.requestType
    },
    firstName(state) {
      return state.applicantInfo.applicantname.firstName
    },
    lastName(state) {
      return state.applicantInfo.applicantname.lastname
    },
    address(state) {
      return state.applicantInfo.contactInfo.address
    },
    conEmail(state) {
      return  state.applicantInfo.contactInfo.email
    },
    contactName(state) {
      return state.applicantInfo.contactInfo.contactName
    },
    phone(state) {
      return state.applicantInfo.contactInfo.phone
    },
    fax(state) {
      return state.applicantInfo.contactInfo.fax
    },
    jurisdiction(state) {
      return  state.additionalCompInfo.jurisdiction
    },
    natureOfBusiness(state) {
      return state.additionalCompInfo.natureOfBussiness
    },
    nuans(state) {
      return state.additionalCompInfo.nuans
    },
    sk_name(state) {
      return state.additionalCompInfo.sk_name
    },
    nr_status(state) {
      return state.additionalCompInfo.nr_status
    },
    examiner(state) {
      return state.examiner
    },
    priority(state) {
      return state.priority
    },
    reSubmissionYN(state) {
      return state.reSubmission.reSubmissionYN
    },
    linkedNR(state) {
      return state.reSubmission.linkedNR
    },
    reservationCount(state) {
      return state.reservationCount
    },
    expiryDate(state) {
      return state.expiryDate
    },
    issue_Match(state) {
      return state.issue.issue_Match
    },
    issue_Match_Text(state) {
      return state.issue.issue_Match_Text
    },
    issue_Consent(state) {
      return state.issue.issue_Consent
    },
    issue_Consent_Text(state) {
      return state.issue.issue_Consent_Text
    },
    issue_TradeMark(state) {
      return state.issue.issue_TradeMark
    },
    issue_TradeMark_Text(state) {
      return state.issue.issue_TradeMark_Text
    },
    issue_History(state) {
      return state.issue.issue_History
    },
    issue_History_Text(state) {
      return state.issue.issue_History_Text
    },
    issue_Format(state) {
      return state.issue.issue_Format
    },
    issue_Format_Text(state) {
      return state.issue.issue_Format_Text
    },
    details(state) {
      return state.details
    },
    addInfo(state) {
      return state.details
    }
  }
})
