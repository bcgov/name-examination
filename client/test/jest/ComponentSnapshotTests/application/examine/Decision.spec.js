/* eslint-disable */
import { shallowMount } from '@vue/test-utils';
import Decision from '@/components/application/Examine/Decision';
import store from '@/store'

describe('Decision.vue', () => {

    let component;
    beforeEach(() => {
        component = shallowMount(Decision, {store: store});
    });

   it("renders a Decision view", () => {
     expect(component.element).toMatchSnapshot();
   })

  // Add other tests specific to this component and not its sub-components
});
