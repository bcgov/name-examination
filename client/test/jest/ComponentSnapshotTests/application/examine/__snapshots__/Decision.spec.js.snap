// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Decision.vue renders a Decision view 1`] = `
<span>
  <div
    class="row"
  >
    <div
      class="col-6 add-top-padding"
    >
      <h3>
        Conditions
      </h3>
       
      <div>
        <multiselect-stub
          close-on-select="true"
          custom-label="function () { [native code] }"
          deselectgrouplabel="Press enter to deselect group"
          deselectlabel=""
          limit="99999"
          limittext="function(t){return\\"and \\".concat(t,\\" more\\")}"
          maxheight="300"
          multiple="true"
          name="phrase"
          opendirection=""
          options="[object Object]"
          placeholder=""
          selectedlabel=""
          selectgrouplabel="Press enter to select group"
          selectlabel=""
          showlabels="true"
          shownooptions="true"
          shownoresults="true"
          tabindex="0"
          track-by="phrase"
        />
      </div>
    </div>
     
    <div
      class="col-6 add-top-padding"
    >
      <h3>
        Conflicts
      </h3>
       
      <multiselect-stub
        close-on-select="true"
        custom-label="function () { [native code] }"
        deselectgrouplabel="Press enter to deselect group"
        deselectlabel=""
        limit="99999"
        limittext="function(t){return\\"and \\".concat(t,\\" more\\")}"
        max="3"
        maxheight="300"
        multiple="true"
        name=""
        opendirection=""
        placeholder=""
        selectedlabel=""
        selectgrouplabel="Press enter to select group"
        selectlabel=""
        showlabels="true"
        shownooptions="true"
        shownoresults="true"
        tabindex="0"
        track-by="nrNumber"
      />
    </div>
  </div>
   
  <div
    class="row"
  >
    <div
      class="col-6 add-top-padding"
    >
      <h3>
        Generic Rejection Reason (label TBD)
      </h3>
       
      <multiselect-stub
        close-on-select="true"
        deselectgrouplabel="Press enter to deselect group"
        deselectlabel=""
        label="name"
        limit="99999"
        limittext="function(t){return\\"and \\".concat(t,\\" more\\")}"
        maxheight="300"
        multiple="true"
        name=""
        opendirection=""
        placeholder=""
        selectedlabel=""
        selectgrouplabel="Press enter to select group"
        selectlabel=""
        showlabels="true"
        shownooptions="true"
        shownoresults="true"
        tabindex="0"
        track-by="name"
      />
    </div>
     
    <div
      class="col-6 add-top-padding"
    >
      <h3>
        Trademarks
      </h3>
       
      <div>
        <multiselect-stub
          close-on-select="true"
          custom-label="function () { [native code] }"
          deselectgrouplabel="Press enter to deselect group"
          deselectlabel=""
          limit="99999"
          limittext="function(t){return\\"and \\".concat(t,\\" more\\")}"
          maxheight="300"
          multiple="true"
          name="name"
          opendirection=""
          options=""
          placeholder=""
          selectedlabel=""
          selectgrouplabel="Press enter to select group"
          selectlabel=""
          showlabels="true"
          shownooptions="true"
          shownoresults="true"
          tabindex="0"
          track-by="name"
        />
      </div>
    </div>
  </div>
   
  <div
    class="row"
  >
    <div
      class="col add-top-padding"
    />
  </div>
   
  <div
    class="row"
  >
    <div
      class="col add-top-padding"
    >
      <h3>
        Message to Requestor
      </h3>
       
      <div
        class="customer-msg-box"
      >
        <pre>
          
        </pre>
      </div>
       
      <div
        class="character_count"
      >
        
        Character count: 0
        
        <!---->
      </div>
       
      <a
        data-target="#edit-customer-message-modal"
        data-toggle="modal"
        href="#"
        id="edit-customer-message-link"
      >
        Edit...
      </a>
       
      <!---->
    </div>
  </div>
   
  <internalcomments-stub
    layout="decision"
  />
   
  <div
    class="modal fade"
    id="edit-customer-message-modal"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-lg"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title"
            id="exampleModalLabel"
          >
            Edit Customer Comments
          </h5>
           
          <button
            aria-label="Close"
            class="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
        </div>
         
        <div
          class="modal-body"
        >
          <textarea
            class="form-control"
            id="edit-customer-message-textarea"
            rows="10"
          />
        </div>
         
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-sm btn-secondary"
            data-dismiss="modal"
            type="button"
          >
            Cancel
          </button>
           
          <button
            class="btn btn-sm btn-primary"
            type="button"
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
</span>
`;
